### PR TITLE
Fix GitHub actions command injection

### PR DESCRIPTION
**Pull request by dumitory-dev**
# Description

Fix GitHub actions command injection
In the file telegram-bot.yml lines:
```
        elif [[ ! -z "${{ github.event.workflow_run }}" && "${{ github.event.workflow_run.conclusion }}" == "success" ]]; then
          TEXT=`echo -e "<a href=\"${{ env.UNIVERSUM_LOG }}\">Universum run for branch '${{ env.UNIVERSUM_BRANCH }}'</a> <b>SUCCEDED</b>; commit ${{ env.UNIVERUM_COMMIT }} "`
        elif [[ ! -z "${{ github.event.workflow_run }}" && "${{ github.event.workflow_run.conclusion }}" == "failure" ]]; then
          TEXT=`echo -e "<a href=\"${{ env.UNIVERSUM_LOG }}\">Universum run for branch '${{ env.UNIVERSUM_BRANCH }}'</a> <b>FAILED</b>; commit ${{ env.UNIVERUM_COMMIT }} "`
        fi
```
Can be command injection with untrusted env.UNIVERSUM_BRANCH input. See this [PR](https://github.com/Samsung/Universum/pull/841) for example.
An attacker can steal repo secrets that are stored in the env URL variable and maybe GITHUB_TOKEN

This PR fixes this problem with guide from [official github docs](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable)

### CODE REVIEW

File: .github/workflows/telegram-bot.yml [hunk 1]
```diff
@@ -40,42 +40,44 @@ jobs:
...
-          TEXT=`echo -e "<a href=\"${{ env.UNIVERSUM_LOG }}\">Universum run for branch '${{ env.UNIVERSUM_BRANCH }}'</a> <b>FAILED</b>; commit ${{ env.UNIVERUM_COMMIT }} "`
+          ESCAPED_TEXT=`echo -e "$UNIVERSUM_BRANCH"| sed 's/\&/\&amp;/g' | sed 's/</\&lt;/g' | sed 's/>/\&gt;/g'`
+          TEXT=`echo -e "<a href=\"$UNIVERSUM_LOG\">Universum run for branch "$ESCAPED_TEXT"</a> <b>FAILED</b>; commit "$ESCAPED_TEXT" "`
         fi
 
-        if [[ ! -z $TEXT ]]; then
+        if [[ ! -z "$TEXT" ]]; then
           curl --get --data-urlencode "chat_id=${{ secrets.TELEGRAM_CHAT_ID }}" --data-urlencode "disable_web_page_preview=True" \
-          --data-urlencode "text=$TEXT" --data-urlencode "parse_mode=HTML" $URL
+          --data-urlencode "text="$TEXT"" --data-urlencode "parse_mode=HTML" "$URL"
```
Comment (i-keliukh) [inline .github/workflows/telegram-bot.yml:80]: The workflow has failed: https://github.com/Samsung/Universum/actions/runs/9747908907/job/26927384622?pr=842#step:2:67

I think the failure is caused by double usage of quotes: `"text="$TEXT""`

Comment (i-keliukh) [reply .github/workflows/telegram-bot.yml:80]: I am not talking about "Universum check" workflow. The telegram workflow has failed, please see it in the logs of the telegram workflow (unfortunately the github doesn't provide link to the correct failure)

Comment (dumitory-dev) [reply .github/workflows/telegram-bot.yml:80]: Sorry, can you provide failed logs?

I see it:
https://github.com/Samsung/Universum/actions/runs/9756659864
```
{"ok":true,"result":{"message_id":28679,"from":{"id":845688040,"is_bot":true,"first_name":"Universum","username":"universum_github_bot"},"chat":{"id":***,"title":"Universum","type":"group","all_members_are_administrators":true},"date":1719904818,"text":"Universum run for branch 'fix/github-actions-command-injection' FAILED; commit bcb16fefc81e58633caeca5be49de017baef5d1d","entities":[{"offset":0,"length":63,"type":"text_link","url":"https://github.com/Samsung/Universum/actions/runs/9747908907"},{"offset":64,"length":6,"type":"bold"}],"link_preview_options":{"is_disabled":true}}}
```

File: .github/workflows/telegram-bot.yml [hunk 2]
```diff
@@ -40,42 +40,44 @@ jobs:
...
-          TEXT=`echo -e "<a href=\"${{ env.UNIVERSUM_LOG }}\">Universum run for branch '${{ env.UNIVERSUM_BRANCH }}'</a> <b>FAILED</b>; commit ${{ env.UNIVERUM_COMMIT }} "`
+          ESCAPED_TEXT=`echo -e "$UNIVERSUM_BRANCH"| sed 's/\&/\&amp;/g' | sed 's/</\&lt;/g' | sed 's/>/\&gt;/g'`
+          TEXT=`echo -e "<a href=\"$UNIVERSUM_LOG\">Universum run for branch "$ESCAPED_TEXT"</a> <b>FAILED</b>; commit "$ESCAPED_TEXT" "`
         fi
 
-        if [[ ! -z $TEXT ]]; then
+        if [[ ! -z "$TEXT" ]]; then
           curl --get --data-urlencode "chat_id=${{ secrets.TELEGRAM_CHAT_ID }}" --data-urlencode "disable_web_page_preview=True" \
-          --data-urlencode "text=$TEXT" --data-urlencode "parse_mode=HTML" $URL
+          --data-urlencode "text=$TEXT" --data-urlencode "parse_mode=HTML" "$URL"
```
Comment (i-keliukh) [inline .github/workflows/telegram-bot.yml:80]: Now it reports "404 not found": https://github.com/Samsung/Universum/actions/runs/9759162609/job/26937949584?pr=842

Comment (dumitory-dev) [reply .github/workflows/telegram-bot.yml:80]: test

Comment (dumitory-dev) [reply .github/workflows/telegram-bot.yml:80]: test

Comment (dumitory-dev) [reply .github/workflows/telegram-bot.yml:80]: test

**Comment by dumitory-dev** [comment]
test

**Review by i-keliukh — APPROVED** [review]

**Pull request merged**